modules/vmfcore/include/vmf/ireader.hpp

```diff
@@ -64,6 +75,24 @@ class VMF_EXPORT IReader
         std::vector<std::shared_ptr<MetadataInternal>>& metadata ) = 0;
 
     virtual bool parseVideoSegments(const std::string& text, std::vector<std::shared_ptr<MetadataStream::VideoSegment> >& segments) = 0;
+
+protected:
```

Comment:
Actually my idea is that an interface class shouldn't have any fields and non-public stuff (that means implementation details).